ParabolInc/parabol
# chore(teams-limit): avoid showing nagging snackbar if the limit is fixed

If the user fixed its exceeding limit, we need to change `suggestedTier` to `starter` to avoid showing nagging snackbar.

How to test:

- Test on a fresh account with fresh active domain without upgraded teams. (e.g. create new user user@freshdomain123.com)
- Add a feature flag on the org

```
r.db('actionDevelopment').table('Organization')
  .getAll(r.args(['PUT_YOUR_ORG_ID_HERE']), {index: 'id'})
  .update({
    featureFlags: r.row('featureFlags').default([]).setInsert('teamsLimit'),
    tierLimitExceededAt: null,
    scheduledLockAt: null,
    lockedAt: null
});
```

- Uncomment https://github.com/ParabolInc/parabol/blob/99e3106cc942b8b74dea6668487fcc6658bede1e/packages/server/billing/helpers/teamLimitsCheck.ts#L14
- Change values to:
```
const enum Threshold {
  MAX_STARTER_TIER_TEAMS = 1,
  MIN_STICKY_TEAM_MEETING_ATTENDEES = 1,
  MIN_STICKY_TEAM_MEETINGS = 1,
  STARTER_TIER_LOCK_AFTER_DAYS = 0
}
```
- Create at least 2 teams
- Run and complete a meeting for each team
- See usage stats enabled but there is no nagging snackbar (only closable snackbar that will not appear again on page refresh)
- Remove one team, see `tierLimitExceeded` flag is gone
- See that you still can access usage stats, but you still don't see nagging snackbar
- Refresh the page, see that everything is fine
- Upgrade to PRO, see that everything is fine
